src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -597,6 +602,46 @@ private object ReadNestedResourceInline(ODataResourceWrapper resourceWrapper, IE
             DeserializationHelpers.SetCollectionProperty(resource, nestedProperty, value, propertyName);
         }
 
+        private bool TryProcessDeltaNestedResourceSet(IEdmProperty collectionProperty, object resource, ODataResourceSetWrapper resourceSetWrapper, ODataDeserializerContext readContext)
+        {
+            string collectionPropertyName = EdmLibHelpers.GetClrPropertyName(collectionProperty, readContext.Model);
+            IDelta resourceDelta = resource as IDelta;
+
+            if (resourceDelta == null)
+            {
+                return false;
+            }
+
+            if (resourceSetWrapper == null)
+            {
+                return resourceDelta.TrySetPropertyReferencedValue(collectionPropertyName, null);
+            }
+
+            if (resourceSetWrapper.Resources == null)
+            {
+                return false;
+            }
+
+            IEdmStructuredTypeReference structuredElementType = collectionProperty.Type.AsCollection().ElementType().AsStructured();
+
+            Type clrElementType = EdmLibHelpers.GetClrType(structuredElementType, readContext.Model);
+            Type deltaType = typeof(Delta<>).MakeGenericType(clrElementType);
+
+            IEnumerable<string> structuralProperties = GetUpdatableProperties(readContext.Model, structuredElementType);
+
+            IList elementDeltas = (IList)Activator.CreateInstance(typeof(List<>).MakeGenericType(deltaType));
```

Comment:
Do you need a List<Delta<T>> here? Seems like you could avoid reflection with simple IList<Delta> since that's what you're putting in the list. An IList of objects would be even simpler.

Reply:
Good point. I copied this blindly from different part of ODataResourceDeserializer but it seems that typing is not needed here (at least tests are passing just fine).